src/openforms/formio/tests/validation/test_file.py

```diff
@@ -204,35 +250,65 @@ def test_no_temporary_upload(self):
         self.assertEqual(error, _("Invalid URL."))
 
     def test_does_not_match_upload_file_size(self):
-
-        data = SubmittedFileFactory.create()
+        temporary_file_upload = TemporaryFileUploadFactory.create(file_size=10)
+        data = SubmittedFileFactory.create(temporary_upload=temporary_file_upload)
         data["size"] = 0
         data["data"]["size"] = 0
 
-        is_valid, errors = validate_formio_data(DEFAULT_FILE_COMPONENT, {"foo": [data]})
+        is_valid, errors = validate_formio_data(
+            DEFAULT_FILE_COMPONENT,
+            {"foo": [data]},
+            extra_context={"submission": temporary_file_upload.submission},
+        )
 
         self.assertFalse(is_valid)
         error = extract_error(errors["foo"][0], "size")
         self.assertEqual(error.code, "invalid")
         self.assertEqual(error, _("Size does not match the uploaded file."))
 
     def test_does_not_match_upload_name(self):
-
-        data = SubmittedFileFactory.create()
+        temporary_file_upload = TemporaryFileUploadFactory.create()
+        data = SubmittedFileFactory.create(temporary_upload=temporary_file_upload)
         data["originalName"] = "unrelated"
         data["data"]["name"] = "unrelated"
 
-        is_valid, errors = validate_formio_data(DEFAULT_FILE_COMPONENT, {"foo": [data]})
+        is_valid, errors = validate_formio_data(
+            DEFAULT_FILE_COMPONENT,
+            {"foo": [data]},
+            extra_context={"submission": temporary_file_upload.submission},
+        )
 
         self.assertFalse(is_valid)
         error = extract_error(errors["foo"][0], "originalName")
         self.assertEqual(error.code, "invalid")
         self.assertEqual(error, _("Name does not match the uploaded file."))
 
+    @tag("security-30")
+    def test_temporary_file_upload_invalid_submission(self):
+        temporary_file_upload = TemporaryFileUploadFactory.create()
+        data = SubmittedFileFactory.create(temporary_upload=temporary_file_upload)
+
+        # Not giving any extra context, so that `validate_formio_data` will create
+        # a dummy submission for us, which will be != `temporary_file_upload.submission`:
+        is_valid, errors = validate_formio_data(
+            DEFAULT_FILE_COMPONENT,
+            {"foo": [data]},
+        )
```

Comment:
and explicitly created different submission would be better for that purpose instead of relying on implicit shims